app/src/main/java/ch/epfl/swissteam/services/PostActivity.java

```diff
@@ -49,9 +49,13 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_post);
         super.onCreateDrawer(BACK);
 
-        //Retrieve the post from the intent which started this activity
+        // Retrieve the post from the intent which started this activity
         Intent callingIntent = getIntent();
-        post_ = callingIntent.getParcelableExtra(PostAdapter.POST_TAG);
+        if (callingIntent.getParcelableExtra(PostAdapter.POST_TAG) != null) {
+            post_ = callingIntent.getParcelableExtra(PostAdapter.POST_TAG);
+        } else if (callingIntent.getParcelableExtra(TodoListAdapter.TODOPOST_TAG) != null) {
```

Comment:
Why do you need to know which intent is calling? 
If you do not use it, you could put the tag in PostActivity and make the other two activities use that tag.

Reply:
It is all about the return button (as mentionned in the body of my last commit).
Indeed, if the Post activity think it's always coming from Nearby, it will always go back to the Nearby fragment which break the flow of the app.